stellarphot/core.py

```diff
@@ -62,3 +64,307 @@ def __init__(self, gain=1.0 * u.electron / u.adu,
+        A table formatted to match the table_descript formatting information
+        but containing no data.
+
+    dec: `astropy.table.Column`
+        A column of declination values with units of degrees
+
+    id: `astropy.table.Column`
+        A column of object id values (no specific format assumed)
+
+    ra: `astropy.table.Column`
+        A column of right ascension values with units of degrees
+
+    Other attributes may be created as defined by table_descript.
+    """
+
+    def __init__(self, table_descript=None, data=None):
+        # Handle parameters
+        self._table_descript = table_descript
+        self.data = data
+
+        # Confirm a proper table description is passed
+        if (type(self._table_descript) is not np.ndarray):
+            raise TypeError(f"You must provide a 4-column numpy array as table_descript (it is type {type(self._table_descript)}).")
+        elif self._table_descript.shape[1] != 4:
+            raise ValueError(f"table_descript must be a 4-column numpy array (it has {self._table_descript.shape[1]} columns).")
+
+        # Extract appropriate information from input `table_descript`
+        colnames = self._table_descript[:,0].tolist()
+        coltypes = self._table_descript[:,1].tolist()
+        colunits = self._table_descript[:,2].tolist()
+        attrnames = self._table_descript[:,3].tolist()
+
+        # Check that required attributes are in the input attributes
+        required_attr = ['ra', 'dec', 'id']
+        required_units = [u.deg, u.deg, None]
+        for i, this_attr in enumerate(required_attr):
+            if this_attr not in attrnames:
+                raise ValueError(f"Required attribute '{this_attr}' is not defined in your table_descript.")
+            if required_units[i] is not None:
+                if colunits[attrnames.index(this_attr)] != required_units[i]:
+                    raise ValueError(f"Required attribute '{this_attr}' must have units {required_units[i]} (table_descript has '{colunits[attrnames.index(this_attr)]}').")
+
+        # Perform processing once initialized
+        if self.data is None:
+            # Create empty astropy table with right format and data
+            self.data = Table(names=colnames, masked=False, dtype=coltypes, units=colunits)
+        else:
+            # Check the format of the data table matches the table_descript by checking
+            # each column listed in table_descript exists and is he write type and unit.
+            # NOTE: This ignores any columns not in the table_descript, it does not remove them.
+            for i,this_col in enumerate(colnames):
+                # Failure is assumed to be due to bad column value
+                try:
+                    data[this_col]
+                except KeyError:
+                    raise ValueError(f"column {this_col} is not present in input data.")
+                # Check type
+                if data[this_col].dtype != coltypes[i]:
```

Comment:
I'm going to read up on dtypes; I think this might be too restrictive: for example, what if the column is `float32` instead of `float64`?

Maybe the better approach is to always create the table, like on line 147, but then set the column?